wasmtime/_sharedmemory.py

```diff
@@ -0,0 +1,74 @@
+from . import _ffi as ffi
+from ctypes import *
+import ctypes
+from wasmtime import MemoryType, WasmtimeError, Engine, Managed
+from ._store import Storelike
+
+
+
+class SharedMemory(Managed["ctypes._Pointer[ffi.wasmtime_sharedmemory_t]"]):
+    def __init__(self, engine: Engine, ty: MemoryType):
+        """
+        Creates a new shared memory in `store` with the given `ty`
+        """
+
+        sharedmemory = ffi.wasmtime_sharedmemory_t()
+        ptr = POINTER(ffi.wasmtime_sharedmemory_t)()
+        error = ffi.wasmtime_sharedmemory_new(engine.ptr(), ty.ptr(), byref(ptr))
+        if error:
+            raise WasmtimeError._from_ptr(error)
+        self._set_ptr(sharedmemory)
+
+    @classmethod
+    def _from_raw(cls, sharedmemory: ffi.wasmtime_sharedmemory_t) -> "SharedMemory":
+        ty: "SharedMemory" = cls.__new__(cls)
+        ty._set_ptr(sharedmemory)
```

Comment:
This is correct, right?

Reply:
Looks good! I might recommend also adding a runtime type-check [like with traps](https://github.com/bytecodealliance/wasmtime-py/blob/157e307aec8e864159181312ecee4e8c283b3c51/wasmtime/_trap.py#L49-L50) to help protect against unsafety at runtime

Reply:
hmmm... you mean something like:
```python
if not isinstance(sharedmemory, ffi.wasmtime_sharedmemory_t):
   raise TypeError(...)
```
?